### ISSUE TITLE
Add "expect" option

### ISSUE THREAD

**Issue by link2xt** [jung-kurt/caddy-cgi#6]
https://caddyserver.com/docs/fastcgi has `expect` option, but `http.cgi` does not. This makes it necessary to move resources to separate directory `/cgit-resource/`.

I have files `cgit.png`, `cgit.cgi` etc. stored inside `/git/` directory and want to handle `/git/` path with `cgit.cgi`, but serve static files `cgit.css` and `cgit.png` for paths `/git/cgit.css` and `/git/cgit.png`.

For example:
```
cgi {
        match /git
        except /git/cgit.css
        except /git/cgit.png
        exec /web/root/dir/git/cgit.cgi
}
```

**Comment by jung-kurt** [comment]
Good idea, @labdsf.

Is the idea that excepted files simply are not processed by the "cgi" plugin, and that some later module in the plugin chain (typically a static handler but not necessarily) handles it?

**Comment by link2xt** [comment]
Yes. I am not familiar with codebase, maybe it is a good idea to look at fastcgi handler code so there is no unexpected inconsistency in how these options work. `cgi` and `fastcgi` should be as similar as possible IMO.

**Comment by jung-kurt** [comment]
I'll take a look at the fastcgi implementation. 

Would wildcard processing be useful? (For example, *except /git/\*.png*)

**Comment by link2xt** [comment]
@jung-kurt Looks like `fastcgi` processes these paths as wildcards, so yes: https://github.com/mholt/caddy/blob/1125a236eabb61bbccb5b6a1af1a48e39da59a20/caddyhttp/fastcgi/fastcgi.go#L448

**Comment by jung-kurt** [comment]
I've got a hectic couple of weeks, but will get to this when I can.

**Comment by jung-kurt** [comment]
Implemented in d6b3c105f0777d7e90577a2b375745a4b72a386e
